services/packages/client/client.go

```diff
@@ -348,8 +349,119 @@ func (l *listCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interfac
 		}
 		fmt.Fprint(state.Out[r.Index], "Installed Packages\n")
 		for _, pkg := range r.Resp.Packages {
-			// Print the package name, version and repo with some reasonable spacing.
-			fmt.Fprintf(state.Out[r.Index], "%40s %16s %32s\n", pkg.Name, pkg.Version, pkg.Repo)
+			// Print the package name.arch, [epoch]:version-release and repo with some reasonable spacing.
+			na := fmt.Sprintf("%s.%s", pkg.Name, pkg.Architecture)
+			evr := fmt.Sprintf("%s-%s", pkg.Version, pkg.Release)
+			if pkg.Epoch != 0 {
+				evr = fmt.Sprintf("%d:%s", pkg.Epoch, evr)
+			}
+			fmt.Fprintf(state.Out[r.Index], "%40s %16s %32s\n", na, evr, pkg.Repo)
+		}
+	}
+	return retCode
+}
+
+type searchCmd struct {
+	packageSystem string
+	name          string
+	installed     bool
+	latest        bool
+}
+
+func (*searchCmd) Name() string     { return "search" }
+func (*searchCmd) Synopsis() string { return "Search NEVRA of a package" }
+func (*searchCmd) Usage() string {
+	return `search --name=X [--installed] [--latest] [--package_system=P]:
+  Search the current installed packages or the latest packages to be updated on the remote machine. The packages will be displayed in NEVRA version.
+`
+}
+
+func (l *searchCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&l.packageSystem, "package-system", "YUM", fmt.Sprintf("Package system to use(one of: [%s])", strings.Join(shortPackageSystemNames(), ",")))
+	f.StringVar(&l.name, "name", "", "Name of package to search")
+	f.BoolVar(&l.installed, "installed", false, "If true print out installed NEVRA of the package")
+	f.BoolVar(&l.latest, "latest", false, "If true print out latest NEVRA of the package")
+}
+
+func getSearchType(i int) (string, error) {
+	switch i {
+	case int(pb.SearchType_SEARCH_TYPE_INSTALLED):
+		return "Installed Package:", nil
+	case int(pb.SearchType_SEARCH_TYPE_LATEST):
+		return "Latest Package:", nil
+	default:
+		return "", fmt.Errorf("unknown search type")
+	}
+}
+
+func (l *searchCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
+	if f.NArg() != 0 {
+		fmt.Fprintln(os.Stderr, "All options are set via flags")
+		return subcommands.ExitFailure
+	}
+	if l.name == "" {
+		fmt.Fprintln(os.Stderr, "--name must be supplied")
+		return subcommands.ExitFailure
+	}
+	var installed, latest = false, false
+
+	// if we don't set flags for current and latest, both should be set by default
+	if !l.installed && !l.latest {
+		installed, latest = true, true
+	}
+	if l.installed {
+		installed = true
+	}
+	if l.latest {
+		latest = true
+	}
```

Comment:
I think let's remove these vars and just use the flags

Reply:
you are right, I should use the flags